# Applied Data Science Capstone

## This Notebook will be used for the Coursera capstone project

## Table of contents
* [Introduction: Business Problem](#introduction)
* [Data](#data)
* [Methodology](#methodology)
* [Analysis](#analysis)
* [Results and Discussion](#results)
* [Conclusion](#conclusion)

## Introduction: Business Problem <a name="introduction"></a>

In this project, we will try to find an optimal location for a business operation. Specifically, a close friend of mine asked me to help him with finding a good place for opening a **cafe in Athina, Greece**. I live in Athina, so it's easy for me to help him with this issue.

It doesn't matter in which borough in Athina his cafe is going to open as soon as it is in the center of a borough with a lot of similar shops.

We will use our data science powers to cluster boroughs in Athina based on the abovementioned criteria. I hope that my work will help him to choose a pretty good location for his cafe.

## Data <a name="data"></a>

Based on the definition of our problem, the factors that will influence our decision are:
* Number of the existing cafe in the boroughs (we need cafe shop to be the first or second popular shop in a borough center)
* Similar boroughs in Athina (so my friend can choose where he prefers to open his cafe) 

I decided to use the coordinates of Athina boroughs centers and search around them (range < 500m) for shops.

I used data sources as follows:
* Borough centers obtained from 
 http://geodata.gov.gr/maps/?package=7a9cb4c4-f192-4df8-a6b4-1143c80ebc2d&resource=525ed8ae-1f2c-4c14-acb5-14d2d00be4a4&locale=el , which is a formally Greece government site
* number of shops and their type and location in every borough center obtained using **Foursquare API**

### Import Libraries

In [1]:
import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
import numpy as np # library to handle data in a vectorized manner
import json # library to handle JSON files

from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

import folium # map rendering library

### Load excel file and create the dataframe

##### I download this file from http://geodata.gov.gr/maps/?package=7a9cb4c4-f192-4df8-a6b4-1143c80ebc2d&resource=525ed8ae-1f2c-4c14-acb5-14d2d00be4a4&locale=el

In [3]:
df = pd.read_excel("athens coordinates.xlsx")
df.head()

Neighborhood  Latitude  Longitude
0          d. agias paraskeuhs kentro   38.0132    23.8210
1  d. agias paraskeuhs pl. ag. ioannh   38.0055    23.8280
2          d. agias paraskeuhs boreia   38.0183    23.8288
3                d. amarousiou kentro   38.0562    23.8068
4                  d. amarousiou oaka   38.0373    23.7946

##### Check dataframe shape

In [4]:
df.shape

(104, 3)

So now we have all 104 boroughs centers in Athina city and we proceed in the analysis.

## Methodology <a name="methodology"></a>

In this project, we will direct our efforts on detecting borough centers that have high coffee shop density ~500m around the borough center.

In the first step, we have collected the required data, borough centers coordinates.

Accordingly, we collect information about shops around those centers, according to Foursquare categorization.

Then, we cluster boroughs into five clusters.

Next step, we choose the cluster that has a cafe shop as a first or second popular shop.

Finally, we propose one of these boroughs as an optimal place to open a cafe.

## Analysis <a name="analysis"></a>

### Generate maps to visualize neighborhoods of Athina Greece

#### Use geopy library to get the latitude and longitude values of Athina

In [13]:
address = 'Athina City, ATH'

geolocator = Nominatim(user_agent="tor_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Athina City are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Athina City are 38.00423275, 23.72378525842142.


#### Create a map of Athina with neighborhoods superimposed on top

In [16]:
# create map of Athina using latitude and longitude values
map_athina = folium.Map(location=[latitude, longitude], zoom_start=11)

# add markers to map
for lat, lng, neighborhood in zip(df['Latitude'], df['Longitude'], df['Neighborhood']):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_athina)  
    
map_athina

#### Define Foursquare Credentials and Version

In [17]:
CLIENT_ID = 'PKKZ00I0FIFWGJYIT3VFOO3P323YVTK2C4USL2SOZWNXGWXD' # your Foursquare ID
CLIENT_SECRET = '1YNIZ2QUSFHUQZGF5K0CWP5ZKLADZSXV1BVFHDQP0RMZ1MZY' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: PKKZ00I0FIFWGJYIT3VFOO3P323YVTK2C4USL2SOZWNXGWXD
CLIENT_SECRET:1YNIZ2QUSFHUQZGF5K0CWP5ZKLADZSXV1BVFHDQP0RMZ1MZY


### Explore Neighborhoods in Athina

#### Create a function to explore all the neighborhoods in Athina

In [20]:
LIMIT=100
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

#### Write the code to run the above function on each neighborhood and create a new dataframe called *athina_venues*.

In [21]:
athina_venues = getNearbyVenues(names=df['Neighborhood'], latitudes=df['Latitude'], longitudes=df['Longitude'])

d. agias paraskeuhs kentro
d. agias paraskeuhs pl. ag. ioannh
d. agias paraskeuhs boreia
d. amarousiou kentro
d. amarousiou oaka
d. brilhssion
melissia
d. hrakleiou kentro
d. hrakleiou boreia
d. khfisias kentro
d. khfisias bd
nea eruthraia
d. lukobrushs
d. peukhs
d. metamorfoseos
d. neas ionias kentro
d. neas ionias perissos
d. neas filadelfeias kentro
d. neas filadelfeias boreio
d.papagou
d. xolargou
d. pentelhs
d. neas pentelhs
d. filothehs
d. neou cuxikou
d. cuxikou
d. xalandriou poludroso
d. xalandriou kentro
d. xalandriou kato
d. agias barbaras
d. agion anarguron - kamaterou
nea xalkhdona
d. aigaleo
d. iliou
d. peristeriou anthoupolh
d. peristeriou kentro
d. peristeriou ag. antonios
d. peristeriou mpournazi
d. petroupolhs
d. xaϊdariou karaϊskakh
d. xaϊdariou dasos
kamatero
liosia
axarnes
axarnes notia
d. athhnaion an. pathsia
d. athhnaion kato pathsia
d. athhnaion kucelh
d. athhnaion pl attikhs
d. athhnaion sepolia
d. athhnaion biktoria
d. athhnaion metaξourgeio
d. athhnaion keram

#### Check the size of the resulting dataframe

In [22]:
athina_venues.shape

(4282, 7)

In [23]:
athina_venues.head()

Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0  d. agias paraskeuhs kentro                38.0132                  23.821   
1  d. agias paraskeuhs kentro                38.0132                  23.821   
2  d. agias paraskeuhs kentro                38.0132                  23.821   
3  d. agias paraskeuhs kentro                38.0132                  23.821   
4  d. agias paraskeuhs kentro                38.0132                  23.821   

                     Venue  Venue Latitude  Venue Longitude Venue Category  
0            Daddy's bread       38.012745        23.820685         Bakery  
1  WallStreet Food and Bar       38.014651        23.821249     Restaurant  
2                    pet4u       38.013986        23.819567      Pet Store  
3                Starbucks       38.013070        23.819316    Coffee Shop  
4         Creperie Asterix       38.013697        23.822646       Creperie

#### Analyze Each Neighborhood

In [24]:
# one hot encoding
athina_onehot = pd.get_dummies(athina_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
athina_onehot['Neighborhood'] = athina_venues['Neighborhood'] 

# move neighborhood column to the first column
#fixed_columns = [toronto_onehot[['Neighborhood']]] + list(toronto_onehot.columns)
#toronto_onehot = toronto_onehot[fixed_columns]

athina_onehot.head()

ATM  Accessories Store  Adult Boutique  African Restaurant  \
0    0                  0               0                   0   
1    0                  0               0                   0   
2    0                  0               0                   0   
3    0                  0               0                   0   
4    0                  0               0                   0   

   American Restaurant  Art Gallery  Art Museum  Arts & Crafts Store  \
0                    0            0           0                    0   
1                    0            0           0                    0   
2                    0            0           0                    0   
3                    0            0           0                    0   
4                    0            0           0                    0   

   Arts & Entertainment  Asian Restaurant  Athletics & Sports  \
0                     0                 0                   0   
1                     0                 0                   0   
2                     0                 0                   0   
3                     0                 0                   0   
4                     0                 0                   0   

   Auto Dealership  Auto Garage  Auto Workshop  BBQ Joint  Baby Store  \
0                0            0              0          0           0   
1                0            0              0          0           0   
2                0            0              0          0           0   
3                0            0              0          0           0   
4                0            0              0          0           0   

   Bagel Shop  Bakery  Bank  Bar  Baseball Field  Basketball Court  \
0           0       1     0    0               0                 0   
1           0       0     0    0               0                 0   
2           0       0     0    0               0                 0   
3           0       0     0    0               0                 0   
4           0       0     0    0               0                 0   

   Basketball Stadium  Bay  Beach  Beach Bar  Bed & Breakfast  Beer Bar  \
0                   0    0      0          0                0         0   
1                   0    0      0          0                0         0   
2                   0    0      0          0                0         0   
3                   0    0      0          0                0         0   
4                   0    0      0          0                0         0   

   Beer Garden  Beer Store  Betting Shop  Bistro  Board Shop  Bookstore  \
0            0           0             0       0           0          0   
1            0           0             0       0           0          0   
2            0           0             0       0           0          0   
3            0           0             0       0           0          0   
4            0           0             0       0           0          0   

   Bougatsa Shop  Boutique  Bowling Alley  Breakfast Spot  Brewery  \
0              0         0              0               0        0   
1              0         0              0               0        0   
2              0         0              0               0        0   
3              0         0              0               0        0   
4              0         0              0               0        0   

   Bubble Tea Shop  Building  Burger Joint  Bus Station  Bus Stop  Butcher  \
0                0         0             0            0         0        0   
1                0         0             0            0         0        0   
2                0         0             0            0         0        0   
3                0         0             0            0         0        0   
4                0         0             0            0         0        0   

   Café  Camera Store  Candy Store  Cheese Shop  Chinese Restaurant  \
0     0             0            0            0                  

#### Group rows by neighborhood and by taking the mean of the frequency of occurrence of each category

In [25]:
athina_grouped = athina_onehot.groupby('Neighborhood').mean().reset_index()
athina_grouped

Neighborhood       ATM  Accessories Store  \
0                               axarnes  0.000000           0.000000   
1                         axarnes notia  0.000000           0.000000   
2                     d. agias barbaras  0.000000           0.000000   
3            d. agias paraskeuhs boreia  0.000000           0.000000   
4            d. agias paraskeuhs kentro  0.000000           0.000000   
5    d. agias paraskeuhs pl. ag. ioannh  0.000000           0.000000   
6        d. agion anarguron - kamaterou  0.000000           0.000000   
7                    d. agiou dhmhtriou  0.000000           0.000000   
8                d. agiou ioannou renth  0.000000           0.000000   
9                            d. aigaleo  0.000000           0.000000   
10                            d. alimou  0.000000           0.000000   
11                   d. alimou kalamaki  0.000000           0.000000   
12                 d. amarousiou kentro  0.000000           0.000000   
13                   d. amarousiou oaka  0.000000           0.000000   
14                           d. amfialh  0.000000           0.000000   
15                      d. arguroupolhs  0.000000           0.000000   
16             d. athhnaion ampelokhpoi  0.000000           0.000000   
17             d. athhnaion an. pathsia  0.000000           0.000000   
18                d. athhnaion biktoria  0.000000           0.000000   
19                d. athhnaion eξarxeia  0.000000           0.000000   
20                   d. athhnaion gkuzh  0.000000           0.000000   
21                d. athhnaion katexakh  0.000000           0.000000   
22            d. athhnaion kato pathsia  0.000000           0.000000   
23                  d. athhnaion kentro  0.000000           0.000000   
24              d. athhnaion kerameikos  0.000000           0.000000   
25                d. athhnaion kolonaki  0.000000           0.000000   
26                 d. athhnaion koukaki  0.000000           0.000000   
27                  d. athhnaion kucelh  0.000000           0.000000   
28            d. athhnaion metaξourgeio  0.000000           0.000000   
29                    d. athhnaion mets  0.000000           0.000000   
30             d. athhnaion monasthraki  0.000000           0.000000   
31                d. athhnaion pagkrati  0.000000           0.000000   
32                d. athhnaion panormou  0.000000           0.000000   
33               d. athhnaion petralona  0.000000           0.000000   
34              d. athhnaion pl attikhs  0.000000           0.000000   
35                 d. athhnaion sepolia  0.000000           0.000000   
36                        d. brilhssion  0.000000           0.000000   
37                           d. buronos  0.000000           0.000000   
38                           d. cuxikou  0.000000           0.000000   
39                            d. dafnhs  0.000000           0.000000   
40                       d. drapetsonas  0.000000           0.000000   
41                         d. ellhnikou  0.000000           0.000000   
42                         d. filothehs  0.000000           0.000000   
43                  d. galatsiou kentro  0.000000           0.000000   
44                 d. galatsiou lampinh  0.000000           0.000000   
45                          d. glufadas  0.000000           0.000000   
46                      d. glufadas ano  0.000000           0.000000   
47                       d. hlioupoleos  0.000000           0.000000   
48                  d. hrakleiou boreia  0.000000           0.000000   
49                  d. hrakleiou kentro  0.000000           0.000000   
50                             d. iliou  0.000000           0.000000   
51                       d. kaisarianhs  0.000000           0.000000   
52                 d. kallitheas boreia  0.000000           0.000000   
53             d. kallitheas tzitzifies  0.000000           0.000000   
54                       d. keratsiniou  0.000000           0.00000

#### Write a function to sort the venues in descending order

In [26]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

#### Create the new dataframe and display the top 10 venues for each neighborhood

In [27]:
num_top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighborhood']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighborhood'] = athina_grouped['Neighborhood']

for ind in np.arange(athina_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(athina_grouped.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted.head()

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
0                     axarnes                  Café     Mobile Phone Shop   
1               axarnes notia          Dessert Shop    Light Rail Station   
2           d. agias barbaras      Greek Restaurant                  Café   
3  d. agias paraskeuhs boreia         Grocery Store             Pool Hall   
4  d. agias paraskeuhs kentro           Coffee Shop     Electronics Store   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0                   Gym                 Plaza                   Pub   
1                 Hotel      Basketball Court           Supermarket   
2         Grocery Store         Jewelry Store                  Park   
3                Bakery                 Plaza      Greek Restaurant   
4                Bakery                  Café           Supermarket   

  6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
0       Meze Restaurant          Cocktail Bar                Bakery   
1                  Café            Food Court          Food Service   
2               Taverna          Optical Shop                Bakery   
3  Gym / Fitness Center      Basketball Court             Bookstore   
4           Pizza Place        Clothing Store              Pharmacy   

  9th Most Common Venue 10th Most Common Venue  
0           Coffee Shop            Supermarket  
1            Food Stand             Food Truck  
2          Fish Taverna            Coffee Shop  
3            Hotel Pool            Gaming Cafe  
4            Baby Store         Cosmetics Shop

### Cluster Neighborhoods

#### Run *k*-means to cluster the neighborhood into 5 clusters

In [28]:
# set number of clusters
kclusters = 5

athina_grouped_clustering = athina_grouped.drop('Neighborhood', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(athina_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:10]

array([3, 1, 3, 1, 1, 3, 2, 1, 4, 2])

#### Create a new dataframe that includes the cluster as well as the top 10 venues for each neighborhood

In [29]:
# add clustering labels
neighborhoods_venues_sorted.insert(0, 'Cluster Labels', kmeans.labels_)

athina_merged = df

# merge athina_grouped with athina_data to add latitude/longitude for each neighborhood
athina_merged = athina_merged.join(neighborhoods_venues_sorted.set_index('Neighborhood'), on='Neighborhood')

athina_merged.head() # check the last columns!

Neighborhood  Latitude  Longitude  Cluster Labels  \
0          d. agias paraskeuhs kentro   38.0132    23.8210               1   
1  d. agias paraskeuhs pl. ag. ioannh   38.0055    23.8280               3   
2          d. agias paraskeuhs boreia   38.0183    23.8288               1   
3                d. amarousiou kentro   38.0562    23.8068               2   
4                  d. amarousiou oaka   38.0373    23.7946               1   

  1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
0           Coffee Shop     Electronics Store                Bakery   
1                  Café      Greek Restaurant                   Bar   
2         Grocery Store             Pool Hall                Bakery   
3                  Café                   Bar                Bakery   
4        Clothing Store                  Café     Electronics Store   

  4th Most Common Venue 5th Most Common Venue 6th Most Common Venue  \
0                  Café           Supermarket           Pizza Place   
1      Sushi Restaurant  Gym / Fitness Center        Tennis Stadium   
2                 Plaza      Greek Restaurant  Gym / Fitness Center   
3         Souvlaki Shop           Coffee Shop                Office   
4        Cosmetics Shop      Basketball Court        Lingerie Store   

  7th Most Common Venue 8th Most Common Venue    9th Most Common Venue  \
0        Clothing Store              Pharmacy               Baby Store   
1                   Pub           Coffee Shop  Grilled Meat Restaurant   
2      Basketball Court             Bookstore               Hotel Pool   
3           Music Venue             Multiplex             Cocktail Bar   
4  Gym / Fitness Center           Coffee Shop              Supermarket   

  10th Most Common Venue  
0         Cosmetics Shop  
1              Nightclub  
2            Gaming Cafe  
3                  Plaza  
4    Sporting Goods Shop

#### Visualize the resulting clusters

In [30]:
#toronto_merged['Cluster Labels'] = toronto_merged['Cluster Labels'].fillna(0.0).astype(int)

In [31]:
# create map
map_clusters = folium.Map(location=[latitude, longitude], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(athina_merged['Latitude'], athina_merged['Longitude'], athina_merged['Neighborhood'], athina_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

### Examine Clusters

#### Cluster 1

In [39]:
athina_merged.loc[athina_merged['Cluster Labels'] == 0, athina_merged.columns[[0] + list(range(3, athina_merged.shape[1]))]]

Neighborhood  Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
21  d. pentelhs               0              Bus Stop                Bakery   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
21            Restaurant                  Food     Food & Drink Shop   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
21            Food Court          Food Service            Food Stand   

   9th Most Common Venue 10th Most Common Venue  
21            Food Truck            Yoga Studio

#### Cluster 2

In [40]:
athina_merged.loc[athina_merged['Cluster Labels'] == 1, athina_merged.columns[[0] + list(range(3, athina_merged.shape[1]))]]

Neighborhood  Cluster Labels    1st Most Common Venue  \
0     d. agias paraskeuhs kentro               1              Coffee Shop   
2     d. agias paraskeuhs boreia               1            Grocery Store   
4             d. amarousiou oaka               1           Clothing Store   
5                  d. brilhssion               1                     Park   
10                d. khfisias bd               1                   Bistro   
11                 nea eruthraia               1             Dessert Shop   
16       d. neas ionias perissos               1              Supermarket   
18   d. neas filadelfeias boreio               1            Souvlaki Shop   
19                     d.papagou               1                     Park   
20                   d. xolargou               1             Cocktail Bar   
22              d. neas pentelhs               1  Grilled Meat Restaurant   
23                  d. filothehs               1                    Plaza   
24               d. neou cuxikou               1       Seafood Restaurant   
25                    d. cuxikou               1                    Plaza   
28            d. xalandriou kato               1              Tailor Shop   
36   d. peristeriou ag. antonios               1            Souvlaki Shop   
42                        liosia               1              Pizza Place   
44                 axarnes notia               1             Dessert Shop   
49          d. athhnaion sepolia               1              Supermarket   
73            d. agiou dhmhtriou               1           Breakfast Spot   
74                     d. alimou               1   Furniture / Home Store   
75            d. alimou kalamaki               1         Asian Restaurant   
92             d. nikaias dutika               1                     Park   
94         d. peiraios maniatika               1              Supermarket   
98          d. peiraios tzabella               1            Souvlaki Shop   
100         d. peiraios peiraϊkh               1       Seafood Restaurant   

    2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
0       Electronics Store                Bakery                  Café   
2               Pool Hall                Bakery                 Plaza   
4                    Café     Electronics Store        Cosmetics Shop   
5       Convenience Store          Betting Shop      Basketball Court   
10            Video Store          Fish Taverna         Auto Workshop   
11     Seafood Restaurant        Clothing Store      Sushi Restaurant   
16                  Plaza                   Pub         Souvlaki Shop   
18     Athletics & Sports            Smoke Shop            Food Truck   
19                 Bakery            Playground          Dessert Shop   
20           Dessert Shop     Electronics Store      Greek Restaurant   
22       Greek Restaurant           Yoga Studio                Bakery   
23                   Park                  Café         Movie Theater   
24          Souvlaki Shop                Bakery          Cocktail Bar   
25                   Park            Playground        Baseball Field   
28            Supermarket                   Spa                  Café   
36           Dessert Shop                  Café           Supermarket   
42           Burger Joint           Supermarket          Dessert Shop   
44     Light Rail Station                 Hotel      Basketball Court   
49             Baby Store                  Café           Pizza Place   
73         Ice Cream Shop      Greek Restaurant     Mobile Phone Shop   
74                   Pool                   Gym                  Park   
75                   Park      Greek Restaurant           Coffee Shop   
92                 Bakery            Steakhouse                  Café   
94             Food Truck     Mobile Phone Shop           Coffee Shop   
98                    Gym     Convenience Store               Taverna   
100          Fish Taverna                 

#### Cluster 3

In [41]:
athina_merged.loc[athina_merged['Cluster Labels'] == 2, athina_merged.columns[[0] + list(range(3, athina_merged.shape[1]))]]

Neighborhood  Cluster Labels 1st Most Common Venue  \
3              d. amarousiou kentro               2                  Café   
6                          melissia               2                   Bar   
9                d. khfisias kentro               2                  Café   
13                        d. peukhs               2                  Café   
14                 d. metamorfoseos               2         Souvlaki Shop   
17      d. neas filadelfeias kentro               2                   Bar   
26          d. xalandriou poludroso               2                Bakery   
27             d. xalandriou kentro               2                   Bar   
30   d. agion anarguron - kamaterou               2                 Plaza   
31                    nea xalkhdona               2                  Café   
32                       d. aigaleo               2                  Café   
35            d. peristeriou kentro               2                  Café   
41                         kamatero               2                  Café   
45         d. athhnaion an. pathsia               2                  Café   
46        d. athhnaion kato pathsia               2  Gym / Fitness Center   
50            d. athhnaion biktoria               2                  Café   
51        d. athhnaion metaξourgeio               2               Theater   
52          d. athhnaion kerameikos               2             Nightclub   
53           d. athhnaion petralona               2       Meze Restaurant   
54             d. athhnaion koukaki               2                  Café   
55         d. athhnaion monasthraki               2                  Café   
57            d. athhnaion pagkrati               2                  Café   
58            d. athhnaion kolonaki               2                  Café   
59            d. athhnaion eξarxeia               2                   Bar   
61         d. athhnaion ampelokhpoi               2                  Café   
62            d. athhnaion panormou               2                  Café   
63            d. athhnaion katexakh               2           Coffee Shop   
64              d. athhnaion kentro               2                  Café   
66              d. galatsiou kentro               2                  Café   
68                        d. dafnhs               2                  Café   
70                      d. zografou               2                  Café   
71                   d. hlioupoleos               2                  Café   
72                   d. kaisarianhs               2                  Café   
76                      d. glufadas               2          Dessert Shop   
78                     d. ellhnikou               2              Pharmacy   
79                  d. arguroupolhs               2                   Bar   
80             d. kallitheas boreia               2           Coffee Shop   
81         d. kallitheas tzitzifies               2             Nightclub   
83                        d. taurou               2                  Café   
84                  d. neas smurnhs               2                  Café   
85               d. palaiou falhrou               2          Dessert Shop   
86                   d. keratsiniou               2                  Café   
89                    d. korudallou               2                  Café   
90                d. korudallou ano               2                  Café   
91                       d. nikaias               2      Greek Restaurant   
95              d. peiraios kaminia               2           Snack Place   
97             d. peiraios kastella               2    Seafood Restaurant   
99            d. peiraios freattuda               2                   Bar   
101           d. peiraios tercithea               2                   Bar   
102          d. peiraios hlektrikos               2           Snack Place   
103                    d. peramatos               2       Harbor / Marina   

    2nd Most Commo

#### Cluster 4

In [42]:
athina_merged.loc[athina_merged['Cluster Labels'] == 3, athina_merged.columns[[0] + list(range(3, athina_merged.shape[1]))]]

Neighborhood  Cluster Labels 1st Most Common Venue  \
1   d. agias paraskeuhs pl. ag. ioannh               3                  Café   
7                  d. hrakleiou kentro               3                  Café   
8                  d. hrakleiou boreia               3                  Café   
12                       d. lukobrushs               3      Greek Restaurant   
15               d. neas ionias kentro               3                  Café   
29                   d. agias barbaras               3      Greek Restaurant   
33                            d. iliou               3                  Café   
34           d. peristeriou anthoupolh               3                  Café   
37            d. peristeriou mpournazi               3                  Café   
38                      d. petroupolhs               3                  Café   
39             d. xaϊdariou karaϊskakh               3                  Café   
40                  d. xaϊdariou dasos               3                  Café   
43                             axarnes               3                  Café   
47                 d. athhnaion kucelh               3                  Café   
48             d. athhnaion pl attikhs               3                  Café   
56                   d. athhnaion mets               3                  Café   
60                  d. athhnaion gkuzh               3                  Café   
65                          d. buronos               3                  Café   
67                d. galatsiou lampinh               3                  Café   
69                          d. umhttou               3                  Café   
77                     d. glufadas ano               3       Meze Restaurant   
82                        d. mosxatou                3      Greek Restaurant   
87                          d. amfialh               3                  Café   
88                      d. drapetsonas               3      Greek Restaurant   
96              d. peiraios neo falhro               3        Soccer Stadium   

   2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
1       Greek Restaurant                   Bar      Sushi Restaurant   
7       Greek Restaurant               Brewery           Snack Place   
8          Souvlaki Shop          Dance Studio          Dessert Shop   
12           Supermarket                Bakery           Coffee Shop   
15      Greek Restaurant            Shoe Store                   Gym   
29                  Café         Grocery Store         Jewelry Store   
33      Greek Restaurant           Coffee Shop       Meze Restaurant   
34         Jewelry Store               Taverna                   Gym   
37      Greek Restaurant       Meze Restaurant             Nightclub   
38                   Bar           Supermarket             Nightclub   
39      Greek Restaurant       Meze Restaurant          Dessert Shop   
40                   Bar                Bakery          Soccer Field   
43     Mobile Phone Shop                   Gym                 Plaza   
47      Greek Restaurant         Souvlaki Shop          Dessert Shop   
48                Bakery           Snack Place      Greek Restaurant   
56                   Bar          Cocktail Bar                   Gym   
60      Greek Restaurant                Bakery           Supermarket   
65                 Plaza                  Park      Greek Restaurant   
67      Greek Restaurant           Coffee Shop           Supermarket   
69      Greek Restaurant       Meze Restaurant                 Plaza   
77                  Café  Gym / Fitness Center                Bakery   
82                  Café       Meze Restaurant           Supermarket   
87         Souvlaki Shop           Coffee Shop           Supermarket   
88          Burger Joint           Pizza Place     Convenience Store   
96                  Café          Soccer Field      Greek Restaurant   

   5th Most Common Venue    6th Most Common Venue 7th Most Common Venue 

#### Cluster 5

In [43]:
athina_merged.loc[athina_merged['Cluster Labels'] == 4, athina_merged.columns[[0] + list(range(3, athina_merged.shape[1]))]]

Neighborhood  Cluster Labels 1st Most Common Venue  \
93  d. agiou ioannou renth               4          Soccer Field   

   2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
93               Stadium          Tennis Court       Warehouse Store   

   5th Most Common Venue 6th Most Common Venue 7th Most Common Venue  \
93                  Park    Athletics & Sports            Food Stand   

   8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
93                  Food     Food & Drink Shop             Food Court

## Results and Discussion <a name="results"></a>

Our analysis shows that cluster number 4 is the one with a high density of cafe shops, so these borough centers are candidates' places for our proposal.

We will take only those centers that have a cafe shop as the first common venue.

That choice narrows our results furthermore from 25 to 19 centers. 

Lastly, if we want something closer to Athina center, we have only 6 centers.

We can propose any one of those 6 for opening a cafe shop.

## Conclusion <a name="conclusion"></a>

The purpose of this project was to find for a friend of mine an optimal location for a business operation in Athina Greece, specifically a cafe shop. As my friend request, it doesn't matter in which borough in Athina his cafe is going to open as soon as it is in the center of a borough with a lot of similar shops. Our initial data includes 104 borough centers and after analysis, we cluster the centers into 5 clusters, we choose cluster number 4 as the most suitable and finally, we narrow down into 6 centers as optimal locations.
Of course, can be done more work and search these areas with more details.   